### ISSUE TITLE
Revoked token is still valid for `UseLocalServer()` configuration

### ISSUE THREAD

**Issue by dulibanarkadiusz** [openiddict/openiddict-core#2073]
### Confirm you've already contributed to this project or that you sponsor it

- [X] I confirm I'm a sponsor or a contributor

### Version

5.5.0

### Describe the bug

Hello,
In some places of my application (e.g. logout endpoint) I perform a user token invalidation operation. It's easy to achieve using embedded method:
`OpenIddictTokenManager.RevokeByAuthorizationIdAsync("74ebcdcf-2b11-422b-9efa-812afde2151e");`

This way works 100% valid for introspection mechanism, however for `UseLocalServer()` something needs an improvement. The local server seems to store most of its information in cache. The application will remain insensitive to what is done with the token in the database. 
Calling the token revocation method does not invalidate the token, the user can still perform operations in the application - as long as the access token has not exceeded its expiration date. This is inconsistent - using the introspection mechanism, tokens expire immediately after they are revoked and can no longer be used.

In my opinion, the UseLocalServer() method should update the application memory to invalidate the tokens immediately as well. Then the behavior of the package would be consistent - whether I use introspection or not.

### To reproduce

```
services
.AddOpenIddict()
.AddCore(options =>
{
	options
		.UseEntityFrameworkCore()
		.UseDbContext<ApplicationDbContext>();
})
.AddServer(options =>
{
	// ...
})
.AddValidation(options =>
{
	options.UseLocalServer();
	options.UseSystemNetHttp();
	options.UseAspNetCore();
});
```

**Comment by kevinchalet** [comment]
Hi @dulibanarkadiusz,

GitHub doesn't recognize you as a sponsor. Did you use a different/corporate account to sponsor the project?

**Comment by dulibanarkadiusz** [comment]
Try to check again now 😉

**Comment by kevinchalet** [comment]
👍🏻 

The behavior you're seeing is 100% expected and documented here:
  - https://documentation.openiddict.com/configuration/authorization-storage.html#enabling-authorization-entry-validation-at-the-api-level
  - https://documentation.openiddict.com/configuration/token-storage.html#enabling-token-entry-validation-at-the-api-level

TL;DR:

https://github.com/openiddict/openiddict-core/blob/87715ec4f52b7ef54d12b6cf64f1c9793d2916a9/sandbox/OpenIddict.Sandbox.AspNetCore.Server/Startup.cs#L191-L197

**Comment by kevinchalet** [comment]
Doing some housecleaning, but feel free to reopen if you need additional details.
